test/support/app.ts

```diff
@@ -1,20 +1,33 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import { App as AppUnderTest } from '@ukef/app';
 import { MainModule } from '@ukef/main.module';
+import GraphClientService from '@ukef/modules/graph-client/graph-client.service';
+
+import { MockGraphClientService } from './mocks/graph-client.service.mock';
 
 export class App extends AppUnderTest {
-  static async create(): Promise<App> {
+  mockGraphClientService: MockGraphClientService;
+  static async create(): Promise<MockApp> {
+    const mockGraphClientService = new MockGraphClientService();
+
     const moduleFixture: TestingModule = await Test.createTestingModule({
       imports: [MainModule],
-    }).compile();
+    })
+      .overrideProvider(GraphClientService)
+      .useValue(mockGraphClientService)
+      .compile();
 
     const nestApp = moduleFixture.createNestApplication();
 
     const app = new App(nestApp);
 
     await nestApp.init();
 
-    return app;
+    return { app, mockGraphClientService };
+  }
+
+  getGraphClientServiceMock(): MockGraphClientService {
+    return this.mockGraphClientService;
   }
```

Comment:
is this used?

Reply:
Ah -- No sorry! This was an approach that I later decided against, instead choosing to pass out `mockGraphClientService` in the return value when creating the test app